File: typescript/src/api/types/JudgmentTarget.ts

export interface JudgmentTarget {
    accountId: string;
    conversationId?: (string | null) | undefined;
    kind: JudgmentTarget.Kind;
    messageId?: (string | null) | undefined;
    spanId?: (string | null) | undefined;
}

export namespace JudgmentTarget {
    export const Kind = {
            Message: "message",
            Span: "span"
        } as const;
    export type Kind = typeof Kind[keyof typeof Kind];
}

File: ruby/lib/Sikaru/types/judgment_target.rb
# frozen_string_literal: true

module Sikaru
  module Types
    class JudgmentTarget < Internal::Types::Model
      field :account_id, -> { String }, optional: false, nullable: false, api_name: "accountId"

      field :conversation_id, -> { String }, optional: true, nullable: false, api_name: "conversationId"

      field :kind, -> { Sikaru::Types::JudgmentTargetKind }, optional: false, nullable: false

      field :message_id, -> { String }, optional: true, nullable: false, api_name: "messageId"

      field :span_id, -> { String }, optional: true, nullable: false, api_name: "spanId"
    end
  end
end
